### PR TITLE
[WIP] Add some methods to agent

### PR DESCRIPTION
**Pull request by LudvigHz**
- [x] Add methods for communicating with led board
- [x] Add method for polling keypad controller
- [x] handle password storage and validation

### CODE REVIEW

File: keypad/finite_state_machine/rule.py [hunk 1]
```diff
@@ -75,3 +75,10 @@ def signal_is_square(signal: str) -> bool:
     if signal == "#":
         return True
     return False
+
+
+def signal_is_override(signal: str) -> bool:
+    """ Returns true if signal is the override_signal """
+    if signal == "Y":
```
Comment (LudvigHz) [inline keypad/finite_state_machine/rule.py]: This should be a constant, and not hardcoded in several places

File: keypad/finite_state_machine/finite_state_machine.py [hunk 2]
```diff
@@ -1,8 +1,28 @@
 """Module contains class for the finite state machine"""
 
-from keypad.finite_state_machine.rule import STATES, Rule
+from keypad.finite_state_machine.rule import *
 from keypad.kpc_agent import KPCAgent
 
+RULES = [
+    Rule(STATES.INIT, signal_is_anything, STATES.READ, KPCAgent.init_passcode_entry),
```
Comment (LudvigHz) [inline keypad/finite_state_machine/finite_state_machine.py]: Does it make sense to use the class itself and not the instance connected to the fsm? Maybe make this a function that takes the instance as an argument, and call this in the fsm init.

Comment (LudvigHz) [reply keypad/finite_state_machine/finite_state_machine.py]: Also, the `READ`, `READ2`, `READ3` are a bit confusing. Maybe change the names to something a bit more explanatory? @stianbm

File: keypad/kpc_agent.py [hunk 3]
```diff
@@ -1,33 +1,106 @@
...
+    def compare_new_passwords(self):
+        """ Check that the new passwords match """
+        if self._current_password == self._temp_password:
+            self.update_passcode_file()
+            # TODO add signal to show user if passwords were equal
+
+    def cache_first_password(self):
+        """ Cache first password in order to compare passwords later """
+        self._temp_password = self._current_password
+        self._current_password = []
```
Comment (LudvigHz) [inline keypad/kpc_agent.py:71]: I'm guessing these two are the password and confirm password values. if so, change the names to something a bit clearer

**Comment by LudvigHz** [comment]
@stianbm As far as I know, defining state variables outside of methods (without `self`) only sets them as class variables and not instance variables. In the `Rule` class, the state is initialized like this and later set in methods, with `self.{something}`. I don't think the initialization here affects the variables actually used. Correct me if I'm wrong!

File: keypad/finite_state_machine/finite_state_machine.py [hunk 4]
```diff
@@ -1,8 +1,28 @@
 """Module contains class for the finite state machine"""
 
-from keypad.finite_state_machine.rule import STATES, Rule
+from keypad.finite_state_machine.rule import *
```
Comment (LudvigHz) [inline keypad/finite_state_machine/finite_state_machine.py]: Don't use star imports, it makes it hard to see where stuff comes from.

**Comment by SmithPeder** [comment]
LGTM 🚀

**Pull request merged**